src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -584,6 +584,21 @@ public Integer lockAllRepositories(Long examId) {
         return programmingExercises.size();
     }
 
+    /**
+     * Sets exam exercise transient properties for different exercise types
+     * @param exam - the exam for which we set the exercise properties
+     */
+    public void setExamExerciseProperties(Exam exam) {
+        exam.getExerciseGroups().forEach(exerciseGroup -> {
+            exerciseGroup.getExercises().forEach(exercise -> {
+                // Set transient property for quiz exam exercise if test runs exist
+                if (exercise instanceof QuizExercise) {
```

Comment:
I guess we could set this value for other exercises as well, might be interesting in the future as well 👍